config/config.go

```diff
@@ -419,6 +425,28 @@ func (s *HTTPProbe) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		}
 	}
 
+	if !s.UseHTTP3 {
+		for _, version := range s.ValidHTTPVersions {
+			if version == "HTTP/3.0" {
+				return errors.New("HTTP/3 cannot be used as a valid HTTP version when enable_http3 is false")
```

Comment:
I see two tests for `error parsing config file: when enable_http3 is true, enable_http2 must be set to false` but no test for this error, can you add that as well?